daemon/monitor.go

```diff
@@ -86,6 +86,20 @@ func (daemon *Daemon) StateChanged(id string, e libcontainerd.StateInfo) error {
 		} else {
 			logrus.Warnf("Ignoring StateExitProcess for %v but no exec command found", e)
 		}
+	case libcontainerd.StateAsyncStart:
+		// Start instigated asynchronously, such as in the case of restart on Windows.
+		// The container is not locked in this case
+		c.Lock()
+		defer c.Unlock()
+		c.SetRunning(int(e.Pid), e.State == libcontainerd.StateStart)
```

Comment:
Yes, true. Copy/paste without reading the code from the StateStart case, not thinking that that also covered StateRestore. Updated.